Sources/Instrumentation/URLSession/URLSessionLogger.swift

```diff
@@ -147,7 +147,7 @@ class URLSessionLogger {
         }
         instrumentation.configuration.injectCustomHeaders?(&request, span)
         var instrumentedRequest = request
-        objc_setAssociatedObject(instrumentedRequest, &URLSessionInstrumentation.instrumentedKey, true, .OBJC_ASSOCIATION_COPY_NONATOMIC)
```

Comment:
did it show as a warning? what are your warning settings?

Reply:
<img width="529" src="https://github.com/open-telemetry/opentelemetry-swift/assets/947205/2e70e034-d7a4-46f4-9e73-b2c933f6f58c">

Warning settings are the default by Xcode for Swift packages. But it might be that this warning is new, because I'm already using Xcode 15 beta.

Reply:
We should not use any non production ready Xcode version except for testing, we can expect bug and issues in it. In this case it is suspicious to me, beacuse `objc_setAssociatedObject` documentation is quite clear about the opposite